src/System.Text.Encoding/tests/Encoder/EncoderConvert2.cs

```diff
@@ -188,6 +188,145 @@ public void PosTest9()
+            byte[] bytes = new byte[c_SIZE_OF_ARRAY];
+            Encoder encoder = Encoding.ASCII.GetEncoder();
+
+            for (int i = 0; i < chars.Length; ++i)
+            {
+                chars[i] = _generator.GetChar(-55);
+            }
+
+            int charsUsed;
+            int bytesUsed;
+            bool completed;
+            encoder.Convert(chars, 0, chars.Length, bytes, 0, bytes.Length, false, out charsUsed, out bytesUsed, out completed);
+
+            // set flush to true and try again
+            encoder.Convert(chars, 0, chars.Length, bytes, 0, bytes.Length, true, out charsUsed, out bytesUsed, out completed);
+        }
+        
+        // PosTest11: Call Convert to convert a ASCII character array with ASCII encoder
+        [Fact]
+        public void PosTest11()
+        {
+            char[] chars = "TestLibrary.TestFramework.BeginScenario".ToCharArray();
+            byte[] bytes = new byte[chars.Length];
+            Encoder encoder = Encoding.ASCII.GetEncoder();
+
+            VerificationHelper(encoder, chars, 0, chars.Length, bytes, 0, bytes.Length, false, chars.Length, chars.Length, true, "011.1");
+            VerificationHelper(encoder, chars, 0, chars.Length, bytes, 0, bytes.Length, true, chars.Length, chars.Length, true, "011.2");
+            VerificationHelper(encoder, chars, 0, 0, bytes, 0, 0, true, 0, 0, true, "011.3");
+        }
+        
+        // PosTest12: Call Convert to convert partial of a ASCII character array with ASCII encoder
+        [Fact]
+        public void PosTest12()
+        {
+            char[] chars = "TestLibrary.TestFramework.BeginScenario".ToCharArray();
+            byte[] bytes = new byte[chars.Length];
+            Encoder encoder = Encoding.ASCII.GetEncoder();
+
+            VerificationHelper(encoder, chars, 0, 1, bytes, 0, 1, false, 1, 1, true, "012.1");
+            VerificationHelper(encoder, chars, 0, 1, bytes, 0, 1, true, 1, 1, true, "012.2");
+            VerificationHelper(encoder, chars, 1, 1, bytes, 0, 1, false, 1, 1, true, "012.3");
+            VerificationHelper(encoder, chars, 1, 1, bytes, 0, 1, true, 1, 1, true, "012.4");
+            VerificationHelper(encoder, chars, 1, 1, bytes, 1, 1, false, 1, 1, true, "012.5");
+            VerificationHelper(encoder, chars, 1, 1, bytes, 1, 1, true, 1, 1, true, "012.6");
+
+            // Verify maxBytes is large than character count
+            VerificationHelper(encoder, chars, 0, chars.Length - 1, bytes, 0, bytes.Length, false, chars.Length - 1, chars.Length - 1, true, "012.7");
+            VerificationHelper(encoder, chars, 1, chars.Length - 1, bytes, 0, bytes.Length, true, chars.Length - 1, chars.Length - 1, true, "012.8");
+        }
+        
+        // PosTest13: Call Convert to convert partial of a Unicode character array with ASCII encoder
+        [Fact]
+        public void PosTest13()
+        {
+            char[] chars = "😁Test".ToCharArray();
```

Comment:
Maybe you should use `\u` or `\U` syntax instead of writing the characters like this. Some editors don't handle weird chars well, e.g. sometimes VSCode changes them to �.